src/lib/util/chdcd.cpp

```diff
@@ -1129,6 +1134,437 @@ chd_error chdcd_parse_cue(const char *tocfname, cdrom_toc &outtoc, chdcd_track_i
 	return CHDERR_NONE;
 }
 
+/*---------------------------------------------------------------------------------------
+    chdcd_is_gdicue - determine if CUE contains Redump multi-CUE format for Dreamcast GDI
+----------------------------------------------------------------------------------------*/
+
+/**
+ * Dreamcast GDI has two images on one disc, SINGLE-DENSITY and HIGH-DENSITY. 
+ *
+ * Redump stores both images in a single .cue with a REM comment separating the images.
+ * This multi-cue format replaces the old flawed .gdi format.
+ *
+ *    http://forum.redump.org/topic/19969/done-sega-dreamcast-multicue-gdi/
+ *
+ * This function looks for strings "REM SINGLE-DENSITY AREA" & "REM HIGH-DENSITY AREA"
+ * indicating the Redump multi-cue format and therefore a Dreamcast GDI disc.
+ */
+
+bool chdcd_is_gdicue(const char *tocfname) 
+{
+	FILE *infile;
+	bool has_rem_singledensity = false;
+	bool has_rem_highdensity = false;
+	std::string path = std::string(tocfname);
+
+	infile = fopen(tocfname, "rt");
+	path = get_file_path(path);
+	if (infile == (FILE *)nullptr)
+	{
+		return false;
+	}
+
+	while (!feof(infile))
+	{
+		fgets(linebuffer, 511, infile);
+
+		/* if EOF didn't hit, keep going */
+		if (!feof(infile))
+		{
+			if (!strncmp(linebuffer, "REM SINGLE-DENSITY AREA", 23)) {
+				has_rem_singledensity = true;
+			}
+
+			if (!strncmp(linebuffer, "REM HIGH-DENSITY AREA", 21)) {
+				has_rem_highdensity = true;
+			}
+		}
+	}
```

Comment:
Please be consistent with your formatting.  This is an Allman-style file – it has the opening brace on a separate line for control statements.  Please follow this consistently (or just don’t use braces when there’s a single statement like this.

Also, you can avoid superfluous string comparisons by structuring the code like this:
```
while (!feof(infile) && (!has_rem_singledensity || !has_rem_highdensity))
{
    fgets(linebuffer, 511, infile);
    if (!feof(infile))
    {
        has_rem_singledensity = has_rem_singledensity || !strncmp(linebuffer, "REM SINGLE_DENSITY AREA", 23);
        has_rem_highdensity = has_rem_highdensity || !strncmp(linebuffer, "REM HIGH_DENSITY AREA", 21);
    }
}
```

Reply:
Thanks. Resolved in 91af4f306760585ffc48a0bf302144cd81823cd8